### ISSUE TITLE
Potential Issue With promise.timeout()

### ISSUE THREAD

**Issue by jefflage** [cujojs/when#340]
I apologize in advance for the vague generality of this post.  

We've recently starting using `timeout()` in our app in production.  We've got a couple of fairly complicated operations that have to coordinate the work of multiple external services.  `timeout()` seemed a great way to be able to abort and notify on which external service was falling down.

One of the external services were coordinating with is a WebDAV file server.  We are making requests to that file server using the `request` npm module.  we are piping the responses from that call back to the user's http request object.  we placed set `.timeout(1000)` on the calls to the WebDAV server. 

this works well on a clean start of node.  at some point though, many of those calls start timing out.  bouncing the node process stops the timeouts.  so it seems that at some point either `request` is building up some cruft thats causing it to slow down or `when` is erroneously timing out the calls.

have you seen any issues around `timeout()`?  do you have any suggestions for any special things we should do with promises that timeout?

**Comment by briancavalier** [comment]
@jefflage Have you had any more insights on this?  ie. potential issues with `request` that you mentioned?

**Comment by jefflage** [comment]
No.  It still happens occasionally and we simply restart node when it happens.  I will certainly ping you when we get around to bypassing request to see if it still happens.
